src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -381,5 +394,167 @@ private static void Free(bool ignoreCase, HashSet<string> identifiers, HashSet<s
             SharedPools.StringHashSet.ClearAndFree(identifiers);
             SharedPools.StringHashSet.ClearAndFree(escapedIdentifiers);
         }
+
+        private static void ProcessStringLiteralForGlobalAttributeInfo(
+            SyntaxToken stringLiteral,
+            ISyntaxFacts syntaxFacts,
+            PooledDictionary<string, ArrayBuilder<int>> globalAttributeInfoBuilder)
+        {
+            var text = stringLiteral.ValueText;
+            if (string.IsNullOrEmpty(text) ||
+                !IsTargetOfGlobalAttribute(stringLiteral, syntaxFacts))
+            {
+                return;
+            }
+
+            // Target string must contain a valid symbol DocumentationCommentId,
+            // with an optional '~' prefix.
+            // First walk past this optional prefix.
+            var docCommentId = text[0] == '~' ? text.Substring(1) : text;
+
+            // The first part of the string identifies the kind of member being documented,
+            // via a single character followed by a colon.
+            if (docCommentId.Length < 3 ||
+                !char.IsLetter(docCommentId[0]) ||
+                docCommentId[1] != ':')
+            {
+                return;
+            }
+
+            using var _ = PooledStringBuilder.GetInstance(out var stringBuilder);
+
+            // Now we add entries for all the containing symbols of the target symbol,
+            // starting from the outermost to innermost by parsing the documentation comment ID
+            // from start to end.
+            // For example, consider a field 'Field' defined inside a type 'C' in a namespace 'N'.
+            // This field's documentation comment ID is 'F:N.C.Field'.
+            // We process the string from left to right, starting after the ':' char at position 2.
+            var offset = 2;
+
+            // Iteratively find the next container symbol from current offset.
+            // For the field example above, the first invocation to 'TryProcessNextContainer'
+            // will add 'N' to the string builder and have 'newOffset' at the position after 'N.'
+            // The next invocation will have 'N.C' in the string builder 'newOffset' at the position after 'N.C.' and so on.
+            while (TryProcessNextContainer(stringBuilder, docCommentId, offset, syntaxFacts, out var newOffset))
```

Comment:
do we really need to do this work here?  why not just store the locations?   it's cheap and easy to do.  tehn, find-refs can interpret the string literals when actually doign the searching.

Reply:
no longer relevant